src/webgpu/shader/validation/shader_io/locations.spec.ts

```diff
@@ -74,17 +97,39 @@ g.test('type')
   .params(u =>
     u
       .combine('use_struct', [true, false] as const)
-      .combineWithParams(kTestTypes)
+      .combine('type', new Set([...kValidLocationTypes, ...kInvalidLocationTypes]))
       .beginSubcases()
   )
+  .beforeAllSubcases(t => {
+    if (t.params.type === 'f16') {
+      t.selectDeviceOrSkipTestCase('shader-f16');
+    }
+  })
   .fn(t => {
+    if (
+      t.params.use_struct &&
+      (t.params.type.startsWith('texture') || t.params.type.startsWith('sampler'))
+    ) {
+      t.skip('texture, and sampler types can not be placed in structs');
```

Comment:
Always skipping seems unfortunate, as they'll show up in all the runners. Can we instead filter out the cases so they don't appear in the test runners?

Reply:
I just removed the skips. I'd added them because, in Tint, these failed for other reasons. But, that's a Tint thing, they aren't valid in those locations and should fail, so I think having the test is legitimate.